index.js

```diff
@@ -127,8 +128,17 @@ class Hydra extends EventEmitter {
           this.serviceDescription = this.config.serviceDescription || 'not specified';
           this.serviceVersion = this.config.serviceVersion || 'not specified';
 
+          // if serviceIP field contains a name rather than a dotted IP address
+          // then use DNS to resolve the name to an IP address.
+          if (this.config.serviceIP &&
+              this.config.serviceIP !== '' &&
+              this.config.serviceIP.indexOf('.') === -1) {
```

Comment:
Good point.  So I do some digging around for a regex solution. Then I came across Node's built in support for this: https://nodejs.org/dist/latest-v6.x/docs/api/net.html#net_net_isip_input